be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -714,10 +714,11 @@ std::shared_ptr<ChunksDataRef> ExchangeSinkOperator::construct_brpc_attachment(
     auto chunks_data_ref = std::make_shared<ChunksDataRef>(0);
     for (int i = 0; i < chunk_request->chunks().size(); ++i) {
         auto chunk = chunk_request->mutable_chunks(i);
+        // to reduce peak memory usage
+        chunk->mutable_data()->shrink_to_fit();
         chunk->set_data_size(chunk->data().size());
-
-        chunks_data_ref->data_bytes += chunk->data().size();
-
+        chunks_data_ref->data_bytes += chunk->data().capacity();
+        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
```

Comment:
Are there other ways to get the correct memory usage except calling `CurrentThread::current()`.

Reply:
the key to correct memory usage is getting bytes from string's capacity(), rather than string.size(). besides, `CurrentThread::current()` collects bytes allocated internally, such as allocating inner slots in iobuf, which cann't be seen outside. so the answer is no.